nuclei: add page (#12252)
tldr-pages/tldr: pages/common/nuclei.md

# nuclei

> > Fast and customizable vulnerability scanner based on simple YAML based DSL.

Reviewer: ```suggestion
> Fast and customizable vulnerability scanner based on a simple YAML based DSL.
```

> More information: <https://github.com/projectdiscovery/nuclei>.

> - [u]pdate nuclei [t]emplates to the latest released version:

Reviewer 2: ```suggestion
- [u]pdate `nuclei` [t]emplates to the latest released version:
```

`nuclei -ut`

- List all templates (of protocol type dns):

`nuclei -tl {{-pt dns}}`

- Run automatic web scan using wappalyzer technology detection:

`nuclei -as -u {{scanme.nmap.org}}`

> - Run http protocol type templates of high and critical severity; write results to markdown files:

Reviewer 3: ```suggestion
- Run HTTP [p]rotocol [t]ype templates of high and critical severity, [e]xporting results to [m]arkdown files inside a specific directory:
```

`nuclei -severity high,critical -pt http -u {{http://scanme.sh}} -me {{markdown_dir}}`

> - Run all templates using rate limiting options with reduced silent output:

Reviewer 3: ```suggestion
- Run all templates using a different [r]ate [l]imit and maximum [b]ulk [s]ize with silent output (only showing the findings):
```

> `nuclei -severity high,critical -pt http -u {{http://scanme.sh}} -me {{markdown_dir}}`

Reviewer 2: ```suggestion
`nuclei -severity high,critical -pt http -u {{http://scanme.sh}} -me {{markdown_directory}}`
```

`nuclei -rl {{150}} -bs {{25}} -c {{25}} -silent -u {{http://scanme.sh}}`

> - Run (wordpress) [w]orkflow against a (wordpress) site:

Reviewer 2: ```suggestion
- Run the WordPress [w]orkflow against a WordPress site:
```
How about this?

`nuclei -w {{path/to/nuclei-templates/workflows/wordpress-workflow.yaml}} -u {{https://sample.wordpress.site}}`

> - Run specific template collections from a directory with verbose output in STDERR and detected issues/vulns to a file:
> 
> `nuclei -t {{path/to/nuclei-templates/http}} -u {{http://scanme.sh}} -v -o {{results}}`

Reviewer 3: ```suggestion
- Run one or more specific [t]emplate or directory with [t]emplates with [v]erbose output in `stderr` and [o]utput detected issues/vulns to a file:

`nuclei -t {{path/to/nuclei-templates/http}} -u {{http://scanme.sh}} -v -o {{results}}`
```
I would like to highlight 2 points here:
1. IMO collection sounds like a somewhat special predefined set of templates, which from my interpretation of the README, it only means all templates in the directory:
>    -t, -templates string[]                list of template or template directory to run (comma-separated, file)
2. We use standard notation for some techical terms, which is the case of standard streams, which must be spelled in the following way: `stdout`, `stderr`, `stdin`. See the [style guide](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#special-cases):
> ### Special cases
> 
> - If a command performs irreversible changes to a file system or devices,
>   write every example in a way that cannot be copy pasted thoughtlessly.
>   For example, instead of `ddrescue --force --no-scrape /dev/sda /dev/sdb`
>   write `ddrescue --force --no-scrape {{/dev/sdX}} {{/dev/sdY}}`
>   and use the `{{/dev/sdXY}}` placeholder for *block devices* instead of `/dev/sda1`.
> 
> In general, placeholders should make it as intuitive as possible
> to figure out how to use the command and fill it in with values.
> 
> Technical wording on description lines should use the `backtick` syntax.
> Use backticks on the following:
> 
> - Paths, e.g. `package.json`, `/etc/package.json`.
> - Extensions, e.g. `.dll`.
> - Commands, e.g. `ls`.
> - Standard streams: `stdout`, `stdin`, `stderr`. **Do not** use the full names (e.g. standard output).

> - Run scan based on multiple template conditions:

Reviewer 3: ```suggestion
- Run scan based on one or more [t]emplate [c]onditions:
```

> - Run one or more specific [t]emplate or directory with [t]emplates with [v]erbose output in `stderr` and [o]utput detected issues/vulns to a file:

Reviewer 2: ```suggestion
- Run one or more specific [t]emplate or directory with [t]emplates with [v]erbose output in `stderr` and [o]utput detected issues/vulnerabilities to a file:
```

> - Run one or more specific [t]emplate or directory with [t]emplates with [v]erbose output in `stderr` and [o]utput detected issues/vulnerabilities to a file:

Reviewer 3: ```suggestion
- Run one or more specific [t]emplates or directory with [t]emplates with [v]erbose output in `stderr` and [o]utput detected issues/vulnerabilities to a file:
```
